src/test/ui/issues/issue-40510-3.nll.stderr

```diff
@@ -1,17 +1,15 @@
-error: unsatisfied lifetime constraints
+error: captured variable cannot escape `FnMut` closure body
   --> $DIR/issue-40510-3.rs:18:9
    |
 LL |       || {
-   |       --
-   |       ||
-   |       |return type of closure is [closure@$DIR/issue-40510-3.rs:18:9: 20:10 x:&'2 mut std::vec::Vec<()>]
-   |       lifetime `'1` represents this closure's body
+   |        - inferred to be a `FnMut` closure
 LL | /         || {
 LL | |             x.push(())
 LL | |         }
```

Comment:
Here I suspect this will be a bit confusing because two closures are in play. We actually can probably tell that — if the statement is an assignment with a closure rvalue, then there is an "inner" and an "outer" closure, and maybe we can adjust the message to say something like:

> creates a closure that captures `x` by reference; this inner closure is then returned from the outer closure, allowing `x` to escape the closure body

that wording doesn't feel 💯, though, and I don't know if it's easy to find the name `x` (we don't seem to be using it in other messages)

Maybe something like this?

> creates a closure that contains a reference to a captured variable; this inner closure then escapes the closure body

Reply:
Pushed a change with this.

Reply:
Note that @nikomatsakis was suggesting in particular the phrases "inner closure" and "outer closure", in that there are two closures in play and he wanted some way that the diagnostic would distinguish between the two when discussing them.

I don't see any use of the word "inner" in the diff, so I don't think you *quite* encoded exactly what @nikomatsakis was asking for.

(But I also do think that what you've done is a vast improvement over what we had before...)